sktime/datatypes/_check.py

```diff
@@ -237,18 +237,20 @@ def check_raise(obj, mtype: str, scitype: str = None, var_name: str = "input"):
     ValueError if mtype input argument is not of expected type
     """
     obj_long_name_for_avoiding_linter_clash = obj
-    valid, msg, _ = check_is_mtype(
+    valid = check_is_mtype(
         obj=obj_long_name_for_avoiding_linter_clash,
         mtype=mtype,
         scitype=scitype,
-        return_metadata=True,
+        return_metadata=False,
         var_name=var_name,
     )
 
     if valid:
         return True
     else:
-        raise TypeError(msg)
+        raise TypeError(
+            f"Expected object mtype to be one of {list(mtype)}, was {type(obj)}"
```

Comment:
this message would be confusing, as some mtypes are called like class types, e.g., `pd.DataFrame`.

Reply:
Moved discussion to #4191